File: 01-Backend/DatajudClient.API/DatajudClient.Domain/Models/Tribunais/Tribunal.cs
using DatajudClient.Domain.Models.Endereco;

namespace DatajudClient.Domain.Models.Tribunais
{
    public class Tribunal : ModelBase
    {
        public string Nome { get; set; }
        public string Sigla { get; set; }
        public string Numero { get; set; }
        public string EndpointConsultaNumero { get; set; }
        public int CategoriaId { get; set; }
        public virtual CategoriaTribunal Categoria { get; set; }
        public int EstadoId { get; set; }
        public virtual Estado Estado { get; set; }

        public Tribunal()
        {
            Nome = string.Empty;
            Sigla = string.Empty;
            EndpointConsultaNumero = string.Empty;
            Categoria = new CategoriaTribunal();
            Estado = new Estado();
        }
    }
}

File: 02-Frontend/DatajudClient.Front/src/app/interfaces/tribunal/tribunal.ts
import { Estado } from "../endereco/estado";
import { CategoriaTribunal } from "./categoriaTribunal";

export interface Tribunal {
  id: number;
  nome: string;
  sigla: string;
  numero: string;
  endpointConsultaNumero: string;
  categoria: CategoriaTribunal;
  estado: Estado;
}
